### ISSUE TITLE
Closing media panel does not exit banshee

### ISSUE THREAD

**Issue by nicman23** [arfbtwn/banshee#52]
and if you do not have the notification icon you cannot close with without killing the pid :P

**Comment by arfbtwn** [comment]
I did a bit of testing while producing the patch to fix this and found another workaround - running banshee normally causes the dbus signal to show the main window again :)

Let me know what you think of the patch though and we'll get it merged.

**Comment by nicman23** [comment]
at the time i cannot compile without webkitgtk and the compilation time for it (there are no bin packages for it in arch) is forbidding :(.

**Comment by arfbtwn** [comment]
indeed... you can pass `--disable-webkit` to disable building of those parts temporarily for the test

**Comment by nicman23** [comment]
```
checking for boo >= 0.8.1... no
configure: error: Requires webkit
```
```
./configure --prefix=/usr --sysconfdir=/etc --localstatedir=/var \
              --disable-gst_sharp \
              --enable-gst-native \
              --disable-docs \
              --disable-webkit \
              --disable-static \
              --disable-scrollkeeper \
              --disable-schemas-install \
              --disable-boo \
              --enable-daap \
              --enable-remoteaudio \
              --disable-youtube \
              --disable-gio \
              --disable-gio_hardware \
              --disable-user-help \
              --enable-bpm \
              --enable-coverart \
              --enable-filesystemqueue \
              --enable-fixup \
              --enable-internetarchive \
              --enable-internetradio \
              --enable-lastfm \
              --enable-lastfmstreaming \
              --enable-librarywatcher \
              --enable-mediapanel \
              --enable-minimode \
              --enable-miroguide \
              --enable-mpris \
              --enable-multimediakeys \
              --enable-playqueue \
              --enable-nowplaying \
              --enable-notificationarea \
              --enable-opticaldisc \
              --enable-podcasting \
              --enable-playermigration \
              --with-vendor-build-id=ArchLinux
```

**Comment by arfbtwn** [comment]
Can you paste the full output somewhere? (pastebin, for example) I tried your configure invocation on my system and it was fine - although I used `autogen.sh` instead of the straight configure.

Are you trying this through Arch's version of emerge? - if you know git that would be faster.

**Comment by nicman23** [comment]
sure here https://pastebin.com/0yuA1RgT

it is not emerge just a pkgbuild

**Comment by arfbtwn** [comment]
right... I see - I thought your build was based on `master` branch. The patch for this is merged into `feature/lite` as well but there is a huge change in the build system too so a lot of the configure flags are different. See `./configure --help`.

Update your build script to use `feature/mediapanel` and you'll get `master` plus that single patch.

**Comment by nicman23** [comment]
in the prepare function i have as a temp hack `git checkout feature/lite`

**Comment by arfbtwn** [comment]
`CollectionExtensions` was only added in .NET Core but I don't think building against that will cause any real issues.

Ignore `feature/lite` for now - after configuring and building from `feature/mediapanel` does the patch have the desired effect?

**Comment by nicman23** [comment]
both build but both need the replacement in the previous comment. actually listening to music now with the mediapanel.

it did freeze once but that might be due to my fault, needs more testing. also https://aur.archlinux.org/packages/banshee-git/ if you care add it in the readme.md

**Comment by arfbtwn** [comment]
Merged #59
